docs/products/clickhouse/concepts/indexing.rst

```diff
@@ -1,43 +1,58 @@
 
-.. note:: Even though ClickHouse sorts data by primary key, it is possible `to choose a primary key that is different than the sorting key <https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#choosing-a-primary-key-that-differs-from-the-sorting-key>`_.
+Using sparse indexing has significant consequences for capabilities and limitations of ClickHouse. A primary key as used in ClickHouse do not ensure uniqueness for a single searched item since only every ten thousandth item is indexed. Thus, you need to iterate over thousands of items to find a specific row, which makes this approach inadequate when working with individual rows and suitable for processing millions or trillions of items.
 
-Using sparse indexing has significant consequences for the advantages and limitations of ClickHouse, mainly because such a primary key does not ensure uniqueness for a single searched item. Seeing that we index every ten thousandth item, in order to find a specific individual row we need to iterate over thousands of items. That's why this approach is inadequate when working with individual rows and should be only used to deal when processing millions, or trillions, of items.
+.. topic:: Example
 
-A good use case for it, for example, is analysing error rates based on server log analysis. In such cases we don't  center our attention on individual lines, but look at the overall picture to see trends. Such requests also allow approximate calculations, using only sample of data to make conclusions.
+    When analysing error rates based on a server log analysis, you don't focus on individual lines but look at the overall picture to see trends. Such requests allow approximate calculations using only a sample of data to draw conclusions.
 
-A good primary index should aim at helping limit number of items we need to read in order to process a query.
+A good primary index should help limit the number of items you need to read to process a query.
 
-.. tip:: Find more about ClickHouse `primary indexes in the official documentation of ClickHouse <https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#choosing-a-primary-key-that-differs-from-the-sorting-key>`_.
+.. tip::
+    
+    Learn more about ClickHouse `primary indexes in the official documentation of ClickHouse <https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree/#choosing-a-primary-key-that-differs-from-the-sorting-key>`_.
 
 ClickHouse data skipping indexes
 --------------------------------
 
-**Skipping indexes** are the secondary indexes in ClickHouse. However, they work quite differently to the secondary indexes we are using in other DBMS.
+Skipping indexes are secondary indexes in ClickHouse, but they work quite differently to the secondary indexes used in other DBMSs.
```

Comment:
This line took me a while to understand. Probably consider writing to something like: 

Skipping indexes in ClickHouse function as secondary indexes, but their functionality differs significantly from secondary indexes in other DBMSs.